examples/animation/blend-trees-1d.html

```diff
@@ -0,0 +1,189 @@
+                                "children": [
+                                    {
+                                        "name": "Idle",
+                                        "point": 0.0
+                                    },
+                                    {
+                                        "name": "Walk",
+                                        "point": 1.0
+                                    }
+                                ]
+                            }
+                        }
+                    ],
+                    "transitions": [
+                        {
+                            "from": "START",
+                            "to": "Travel"
+                        },
+                    ]
+                },
+            ],
+            "parameters": {
+                "speed": {
+                    "name": "speed",
+                    "type": "FLOAT",
+                    "value": 0
+                }
+            }
+        };
+
+        var canvas = document.getElementById("application-canvas");
+
+        window.parent.pc = pc;
+
+        // Create the application and start the update loop
+        var app = new pc.Application(canvas);
+
+        // Set the canvas to fill the window and automatically change resolution to be the same as the canvas size
+        app.setCanvasFillMode(pc.FILLMODE_FILL_WINDOW);
+        app.setCanvasResolution(pc.RESOLUTION_AUTO);
+
+        window.addEventListener("resize", function () {
+            app.resizeCanvas(canvas.width, canvas.height);
+        });
+
+        var miniStats = new pcx.MiniStats(app);
+
+        var entity;
+
+        // Create an Entity with a camera component
+        var cameraEntity = new pc.Entity(app);
+        cameraEntity.addComponent("camera", {
+            clearColor: new pc.Color(0, 0, 0)
+        });
+        cameraEntity.translateLocal(0.0, 1.5, 5.0);
+        app.root.addChild(cameraEntity);
+
+        // Create an entity with a light component
+        var light = new pc.Entity();
+        light.addComponent("light", {
+            type: "directional",
+            color: new pc.Color(1, 1, 1),
+            castShadows: true,
+            intensity: 1,
+            shadowBias: 0.2,
+            shadowDistance: 5,
+            normalOffsetBias: 0.05,
+            shadowResolution: 2048
+        });
+        light.setLocalEulerAngles(45, 30, 0);
+        app.root.addChild(light);
+
+        // Load model and animation assets
+        var model = null;
+        var idleAnim = null;
+        var walkAnim = null;
+        var strafeLeftAnim = null;
+        var strafeRightAnim = null;
+        var jogAnim = null;
+
+        var stateGraphAsset = new pc.Asset('stateGraph', 'animstategraph', null, animStateGraph);
+        stateGraphAsset.once('load', function (asset) {
+            var stateGraph = asset.resource;
+        app.assets.loadFromUrl("../assets/models/xbot.glb", "container", function (err, asset) {
```

Comment:
This syntax asks for a better solution. Perhaps small jobs counter with a callback is a better solution to handle this callback pyramid.

Reply:
It does. There's plans in the not too distance future to rewrite these examples. I think that'd be a good time to rethink all of the asset loading for our examples.